snuba/pipeline/single_query_plan_pipeline.py

```diff
@@ -1,58 +1,97 @@
-from snuba.pipeline.query_pipeline import (
-    QueryPipeline,
-    QueryPipelineBuilder,
-)
-from snuba.pipeline.processors import (
-    _execute_entity_processors,
-    _execute_query_plan_processors,
-)
+from abc import ABC
+
 from snuba.datasets.plans.query_plan import (
     ClickhouseQueryPlan,
     ClickhouseQueryPlanBuilder,
     QueryRunner,
 )
+from snuba.pipeline.processors import (
+    ClickhouseProcessorsExecutor,
+    EntityProcessorsExecutor,
+)
+from snuba.pipeline.query_pipeline import (
+    QueryExecutionPipeline,
+    QueryPipelineBuilder,
+    QueryProcessingPipeline,
+)
+from snuba.query.logical import Query as LogicalQuery
 from snuba.request import Request
+from snuba.request.request_settings import RequestSettings
 from snuba.web import QueryResult
 
 
-class SingleQueryPlanPipeline(QueryPipeline):
+class SinglePlanProcessingPipeline(QueryProcessingPipeline, ABC):
     """
-    A query pipeline for a single query plan.
-
-    TODO: Currently only query plan building and processing is done by the query
-    pipeline, eventually the rest of the query processing sequence will move into
-    the pipeline as well.
+    Executes the processing phase of a single plan query. Which means a
+    query based on a single entity, that would produce a query plan based
+    on a single storage.
     """
 
     def __init__(
         self,
-        request: Request,
-        runner: QueryRunner,
         query_plan_builder: ClickhouseQueryPlanBuilder,
-    ):
-        self.__request = request
-        self.__runner = runner
+        processors: ClickhouseProcessorsExecutor,
+    ) -> None:
         self.__query_plan_builder = query_plan_builder
+        self.__entity_processors = EntityProcessorsExecutor()
+        self.__clickhouse_processors = processors
+
+    def execute(
+        self, query: LogicalQuery, settings: RequestSettings
+    ) -> ClickhouseQueryPlan:
+        self.__entity_processors.execute(query, settings)
+
+        query_plan = self.__query_plan_builder.build_plan(query, settings)
+        self.__clickhouse_processors.execute(query_plan, settings)
+        return query_plan
 
-    def execute(self) -> QueryResult:
-        # Execute entity processors
-        _execute_entity_processors(self.__request)
 
-        # Build and execute query plan
-        self.__query_plan = self.__query_plan_builder.build_plan(self.__request)
-        _execute_query_plan_processors(self.__query_plan, self.__request)
-        return self.__query_plan.execution_strategy.execute(
-            self.__query_plan.query, self.__request.settings, self.__runner
+class PreStrategyProcessingPipeline(SinglePlanProcessingPipeline):
```

Comment:
What is the purpose of this class? Why not call `SinglePlanProcessingPipeline(query_plan_builder, ClickhouseProcessorsExecutor(lambda plan: plan.plan_processors))` directly?

Reply:
In the PR above this one I am adding a PostStrategyProcessingPipeline that executes all the query processors and skips the strategy. The issue is that the processing pipeline in a composite query is 75% the same as the single storage one, but then it skips the strategy and just runs the DB query processors. 
Unfortunately if I fit that part in this one as well, this PR will blow up in size.

Reply:
Now we do not have a PreStrategyPipeline of any kind anymore, so this should not apply anymore